

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Homework assignments &mdash; bioinformatics@nau</title>
    
    <link rel="stylesheet" href="../../_static/haiku.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/print.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../',
        VERSION:     '',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../_static/theme_extras.js"></script>
    <link rel="top" title="bioinformatics@nau" href="../../index.html" /> 
  </head>
  <body>
      <div class="header"><h1 class="heading"><a href="../../index.html">
          <span>bioinformatics@nau</span></a></h1>
        <h2 class="heading"><span>Homework assignments</span></h2>
      </div>
      <div class="topnav">
      
        <p>
        <a class="uplink" href="../../index.html">Contents</a>
        </p>

      </div>
      <div class="content">
        
        
  <div class="section" id="homework-assignments">
<h1>Homework assignments<a class="headerlink" href="#homework-assignments" title="Permalink to this headline">¶</a></h1>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<p class="last">I encourage you to discuss homework assignments with each other, but you may not view other student&#8217;s assignments or share your assignment with others. When you start programming, you often think there is a single way to address a task, but that is usually not the case: there are many ways to complete these assignments, and when code has been shared or copied it is often very obvious to a more experienced eye.</p>
</div>
<div class="section" id="turning-in-your-homework-by-email">
<h2>Turning in your homework by email<a class="headerlink" href="#turning-in-your-homework-by-email" title="Permalink to this headline">¶</a></h2>
<p>Your homework must always be turned in with a standardized name. That name should be <tt class="docutils literal"><span class="pre">&lt;nau_id&gt;_&lt;homework_id&gt;.&lt;extension&gt;</span></tt>, where <tt class="docutils literal"><span class="pre">&lt;nau_id&gt;</span></tt> is your NAU identifier (for example, mine is <tt class="docutils literal"><span class="pre">jgc53</span></tt>), and <tt class="docutils literal"><span class="pre">&lt;homework_id&gt;</span></tt> and <tt class="docutils literal"><span class="pre">&lt;extension&gt;</span></tt> are provided on a per-assignment basis.</p>
<p>Unless otherwise noted, homework must be turned in by email to <a class="reference external" href="mailto:jc33&#37;&#52;&#48;nau&#46;edu">jc33<span>&#64;</span>nau<span>&#46;</span>edu</a> before class on the day it is due.</p>
</div>
<div class="section" id="application-presentations">
<h2>Application presentations<a class="headerlink" href="#application-presentations" title="Permalink to this headline">¶</a></h2>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<p class="last">Homework id: <tt class="docutils literal"><span class="pre">app</span></tt>; Extension: <tt class="docutils literal"><span class="pre">pdf</span></tt>; The assignment should be named <tt class="docutils literal"><span class="pre">&lt;group-number&gt;_app.pdf</span></tt> and <tt class="docutils literal"><span class="pre">&lt;group-number&gt;_app_slides.pdf</span></tt>, so for example Group 1&#8217;s assignments would be named <tt class="docutils literal"><span class="pre">group1_app.pdf</span></tt> and <tt class="docutils literal"><span class="pre">group1_app_slides.pdf</span></tt>.</p>
</div>
<div class="section" id="expectations">
<h3>Expectations<a class="headerlink" href="#expectations" title="Permalink to this headline">¶</a></h3>
<p>Each group will be pre-assigned an article seven days before their presentation date. The students will present their article in class the day they&#8217;re assigned. Each member of the group will present part of the material. Answers to the following questions will be turned in (by email, with all group member names included). These answers should form an approximately two-page report.</p>
<blockquote>
<div><ol class="arabic simple">
<li>What is the biological problem that the authors are trying to address?</li>
<li>What is the motivation for addressing this problem?</li>
<li>What previous work has been done in this area? Are there pre-existing tools that address this problem?</li>
<li>What computational technologies did the authors make use of to create this tool (e.g., programming language, databases, etc)?</li>
<li>What preexisting biological resources (e.g., sequence databases) did the authors make use of (if any)?</li>
<li>What is the input to this tool?</li>
<li>What is the output of this tool?</li>
<li>How did the authors test this tool? Was performance benchmarking included in their paper?</li>
<li>How did the authors evaluate whether this tool was giving biologically meaningful results?</li>
</ol>
</div></blockquote>
<p>Presentations will address these same questions, and will additionally include a live demo of the software where the presenters show/discuss the input data, run the application, and show/discuss the output.</p>
</div>
<div class="section" id="grading">
<h3>Grading<a class="headerlink" href="#grading" title="Permalink to this headline">¶</a></h3>
<p>All students in the group will receive the same grade, except in circumstances where it appears that vastly different amounts of effort were contributed by different members of the same group. In addition to being graded by the instructor, anonymous reviews will be turned in by all students - the results of these reviews will factor into each group&#8217;s score.</p>
</div>
<div class="section" id="groups">
<h3>Groups<a class="headerlink" href="#groups" title="Permalink to this headline">¶</a></h3>
<p>Group 1 (3/11/13): jrh355 etb36 rwf25 hhh34 (<a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/early/2010/08/12/bioinformatics.btq461.full.pdf+html">paper</a> and <a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/suppl/2010/08/11/btq461.DC1/supp_mat_rev2.pdf">supplementary material</a> - both are required reading!)</p>
<p>Group 2 (3/11/13): gz38 kn95 sk367 ad572 (<a href="#id4"><span class="problematic" id="id5">`paper &lt;http://dl.dropbox.com/u/2868868/cs299_slides_XCFGcsdFGGad/Genome%20Res.-2009-Parks-1896-904.pdf`_</span></a>)</p>
<p>Group 3 (3/13/13): bs527 eca37 kh832 ajc388 (` &lt;<a class="reference external" href="http://www.mcponline.org/content/5/8/1520.full.pdf+html">http://www.mcponline.org/content/5/8/1520.full.pdf+html</a>&gt;`_ and <a class="reference external" href="http://bmf.colorado.edu/divergentset/">website</a>)</p>
<p>Group 4 (3/13/13): esm23 msk53 pja43 (<a class="reference external" href="http://genomebiology.com/2010/11/8/R86">paper</a>)</p>
</div>
</div>
<div class="section" id="homework-4-tree-of-life-15-points">
<h2>Homework 4: Tree of life (15 points)<a class="headerlink" href="#homework-4-tree-of-life-15-points" title="Permalink to this headline">¶</a></h2>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<p class="last">Homework id: <tt class="docutils literal"><span class="pre">tol</span></tt>; Extension: <tt class="docutils literal"><span class="pre">py</span></tt>, <tt class="docutils literal"><span class="pre">tre</span></tt> and <tt class="docutils literal"><span class="pre">pdf</span></tt>; For this assignment, the files I turn in would be named <tt class="docutils literal"><span class="pre">jgc53_tol.py</span></tt>, <tt class="docutils literal"><span class="pre">jgc53_tol.tre</span></tt> and <tt class="docutils literal"><span class="pre">jgc53_tol.pdf</span></tt>.</p>
</div>
<p>In this assignment you will make use of the PyCogent software package to automate the process of constructing a phylogenetic tree from a set of genes. This will including querying NCBI to obtain sequences, performing a multiple sequence alignment, building a phylogenetic tree, writing a newick string containing that tree to file, and writing a visualization of that tree to a PDF file.</p>
<p>Your script must define a function called <tt class="docutils literal"><span class="pre">obtain_sequences_and_build_tree</span></tt> that takes:
1. a list of queries (as strings) to be run against NCBI;
2. a list of query labels (also as strings) to label the sequences resulting from each query in the final tree;
3. the filepath where the output newick string should be written;
4. the filepath where the output pdf should be written;
5. an optional parameter <tt class="docutils literal"><span class="pre">n</span></tt> which defines how many randomly chosen query results should be chosen for each of the queries. The default value for <tt class="docutils literal"><span class="pre">n</span></tt> should be 5.</p>
<p>Your <tt class="docutils literal"><span class="pre">obtain_sequences_and_build_tree</span></tt> function must return a phylogenetic tree derived from <tt class="docutils literal"><span class="pre">n</span></tt> aligned representatives of each of the queries passed via parameter 1. Your function definition should look exactly like this, where you replace <tt class="docutils literal"><span class="pre">#</span> <span class="pre">do</span> <span class="pre">a</span> <span class="pre">bunch</span> <span class="pre">of</span> <span class="pre">work</span></tt> with your code:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="k">def</span> <span class="nf">obtain_sequences_and_build_tree</span><span class="p">(</span><span class="n">queries</span><span class="p">,</span>
                                    <span class="n">query_labels</span><span class="p">,</span>
                                    <span class="n">output_newick_fp</span><span class="p">,</span>
                                    <span class="n">output_pdf_fp</span><span class="p">,</span>
                                    <span class="n">n</span><span class="o">=</span><span class="mi">5</span><span class="p">):</span>
    <span class="c"># do a bunch of work</span>
    <span class="k">return</span> <span class="n">tree</span>
</pre></div>
</div>
<p>As part of your analysis, you should filter any sequences that have one or more <tt class="docutils literal"><span class="pre">N</span></tt> characters in them. Each sequence label in the output tree should begin with the query label corresponding to that sequence. <tt class="docutils literal"><span class="pre">tree</span></tt> should be a PyCogent <tt class="docutils literal"><span class="pre">PhyloNode</span></tt> object (the output of <tt class="docutils literal"><span class="pre">cogent.app.fasttree.build_tree_from_alignment</span></tt>).</p>
<p>In your script, you should call the function you define as follows:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obtain_sequences_and_build_tree</span><span class="p">(</span>
     <span class="p">[</span><span class="s">&#39;&quot;small subunit rRNA&quot;[ti] AND archaea[orgn]&#39;</span><span class="p">,</span>
      <span class="s">&#39;&quot;small subunit rRNA&quot;[ti] AND bacteria[orgn]&#39;</span><span class="p">,</span>
      <span class="s">&#39;&quot;small subunit rRNA&quot;[ti] AND eukarya[orgn]&#39;</span><span class="p">],</span>
     <span class="p">[</span><span class="s">&#39;A: &#39;</span><span class="p">,</span><span class="s">&#39;B: &#39;</span><span class="p">,</span><span class="s">&#39;E: &#39;</span><span class="p">],</span>
     <span class="s">&quot;&lt;nau-id&gt;_tol.tre&quot;</span><span class="p">,</span>
     <span class="s">&quot;&lt;nau-id&gt;_tol.pdf&quot;</span><span class="p">,</span>
     <span class="n">n</span><span class="o">=</span><span class="mi">5</span><span class="p">)</span>
</pre></div>
</div>
<p>where <tt class="docutils literal"><span class="pre">&lt;nau-id&gt;</span></tt> is replaced with your NAU identifier. This should perform all of the analysis steps and write the newick file and PDF to the directory where you are running the script from. You&#8217;ll turn in the script, the newick file, and the PDF.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last"><a class="reference external" href="http://dl.dropbox.com/u/2868868/pycogent_160dev_docs/cookbook/building_a_tree_of_life.html">This page</a> should help quite a lot.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The cluster has PyCogent, muscle, and FastTree preinstalled. Working there will save you a lot of time on software installation.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Remember that you can call <tt class="docutils literal"><span class="pre">dir()</span></tt> on an object to find out what methods are available to that object. One of the methods associated with your tree object will help you generate a newick formatted tree.</p>
</div>
</div>
<div class="section" id="homework-3-alignments-25-points">
<h2>Homework 3: Alignments (25 points)<a class="headerlink" href="#homework-3-alignments-25-points" title="Permalink to this headline">¶</a></h2>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">This is a big assignment. Start early!</p>
</div>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<p class="last">Homework id: <tt class="docutils literal"><span class="pre">align</span></tt>; Extension: <tt class="docutils literal"><span class="pre">ipynb</span></tt>; For this assignment, the file I turn in would be named <tt class="docutils literal"><span class="pre">jgc53_align.ipynb</span></tt>.</p>
</div>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<p class="last">For this assignment you should work in the QIIME Virtual Box, or in another local IPython installation. You may not use the class IPython Notebook server for this, since it is not a mutli-user environment (i.e, other students will see your work). After installing the QIIME Virtual Box (<a class="reference external" href="http://qiime.org/install/virtual_box.html">instructions here</a>), you can start IPython by opening a terminal and typing <tt class="docutils literal"><span class="pre">ipython</span> <span class="pre">notebook</span></tt>. Leave the terminal window open, and open the URL that is printed to the terminal.</p>
</div>
<p>Begin with the Needleman-Wunsch implementation in the <a class="reference external" href="https://speakerdeck.com/gregcaporaso/bio-299-lecture-8-10-nau-fall-2013">Lecture 10 IPython Notebook</a> and the materials in the <a class="reference external" href="http://nbviewer.ipython.org/4657175/Lecture10.ipynb">Lecture 8-10 slides</a>.</p>
<p>For this assignment you will turn in an IPython notebook. You will generate this notebook by starting with the <a class="reference external" href="http://nbviewer.ipython.org/4657175/Lecture10.ipynb">Lecture 10 IPython Notebook</a> and modifying to add new functionality and annotation.</p>
<div class="section" id="part-1">
<h3>Part 1<a class="headerlink" href="#part-1" title="Permalink to this headline">¶</a></h3>
<p>Add a new function with this <cite>exact</cite> form:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">nw_align</span><span class="p">(</span><span class="n">sequence1</span><span class="p">,</span><span class="n">sequence2</span><span class="p">,</span><span class="n">substitution_matrix</span><span class="p">)</span>
</pre></div>
</div>
<p>This function should return, in this order, the aligned sequence 1 as a string, the aligned sequence 2 as a string, and the score of the global alignment.</p>
<p>To confirm that this is working for you, you should test with the following command, as this is one of the tests that we will apply to your homework:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">nw_align</span><span class="p">(</span><span class="s">&#39;HEAGAWGHEE&#39;</span><span class="p">,</span><span class="s">&#39;PAWHEAE&#39;</span><span class="p">,</span><span class="n">blosum50</span><span class="p">)</span>
</pre></div>
</div>
<p>which should result in the following output:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">(</span><span class="s">&quot;HEAGAWGHE-E&quot;</span><span class="p">,</span> <span class="s">&quot;--P-AW-HEAE&quot;</span><span class="p">,</span> <span class="mf">1.0</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="part-2">
<h3>Part 2<a class="headerlink" href="#part-2" title="Permalink to this headline">¶</a></h3>
<p>In the same notebook, define a new function of the form:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">generate_random_score_distribution</span><span class="p">(</span><span class="n">query_sequence</span><span class="p">,</span><span class="n">subject_sequence</span><span class="p">,</span><span class="n">n</span><span class="p">,</span><span class="n">substitution_matrix</span><span class="p">)</span>
</pre></div>
</div>
<p>Which returns a list of <tt class="docutils literal"><span class="pre">n</span></tt> scores for aligning each of <tt class="docutils literal"><span class="pre">n</span></tt> random sequences of the same length as <tt class="docutils literal"><span class="pre">query_sequence</span></tt> against <tt class="docutils literal"><span class="pre">subject_sequence</span></tt>.</p>
<p>Next, define a function that takes a query sequence, a subject sequence, and a value <tt class="docutils literal"><span class="pre">n</span></tt> with this form:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">fraction_better_or_equivalent_alignments</span><span class="p">(</span><span class="n">query_sequence</span><span class="p">,</span><span class="n">subject_sequence</span><span class="p">,</span><span class="n">n</span><span class="p">,</span><span class="n">substitution_matrix</span><span class="p">)</span>
</pre></div>
</div>
<p>This function should call <tt class="docutils literal"><span class="pre">generate_random_score_distribution</span></tt> to generate a list of scores for random alignments. It should then compute the score for aligning <tt class="docutils literal"><span class="pre">query_sequence</span></tt> against <tt class="docutils literal"><span class="pre">subject_sequence</span></tt>. The return value of this function should be the number of random alignment scores that are better or equal to the actual alignment score divided by <tt class="docutils literal"><span class="pre">n</span></tt>.</p>
<p>After defining this function, use it to compare the following sequences to one another using a value of <tt class="docutils literal"><span class="pre">n=1000</span></tt> when calling <tt class="docutils literal"><span class="pre">fraction_better_or_equivalent_alignments</span></tt> as follows:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">subject</span> <span class="o">=</span> <span class="s">&quot;SAVLDMRPPEITCLCLHSVEWFWATDRAYITKFHVGQPMKCITGCHVFCGPRTSNLLQESCMYCVFSEIGCRNSANCFNFTRSCIRISSYLFSYYIVWGC&quot;</span>
<span class="n">query1</span> <span class="o">=</span> <span class="s">&quot;RHT&quot;</span>
<span class="n">query2</span> <span class="o">=</span> <span class="s">&quot;RHTSWIL&quot;</span>
<span class="n">query3</span> <span class="o">=</span> <span class="s">&quot;RHTSWIIQECWYCWFS&quot;</span>
<span class="n">query4</span> <span class="o">=</span> <span class="s">&quot;RHTSWIIQESCWYCWFSEIGCRNSANWFNFTRSCWRISYLFS&quot;</span>
<span class="n">fraction_better_or_equivalent_alignments</span><span class="p">(</span><span class="n">query1</span><span class="p">,</span><span class="n">subject</span><span class="p">,</span><span class="mi">1000</span><span class="p">,</span><span class="n">blosum50</span><span class="p">)</span>
<span class="o">...</span>
</pre></div>
</div>
<p>Each of these query sequences is designed to be similar to the subject. Also compare some randomly generated query sequences to the subject sequence. Do this several times. In a <em>markdown cell</em> just below this analysis, describe any general patterns that you notice. What do you think this means? Run this example on the alignment we worked through in class (query sequence: <tt class="docutils literal"><span class="pre">HEAGAWGHEE</span></tt>; subject sequence: <tt class="docutils literal"><span class="pre">PAWHEAE</span></tt>) and describe the results. How does this alignment compare to your randomly generated alignments?</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">In the <a class="reference external" href="http://nbviewer.ipython.org/4657175/Lecture8.ipynb">Lecture 8 IPython Notebook</a> there is code illustrating how to generate a random sequence of bases at a given sequence length (see the last cell where <tt class="docutils literal"><span class="pre">root_sequence</span></tt> is defined). Here we&#8217;re working with protein sequences, so the alphabet is different but the process is the same.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">In my <a class="reference external" href="Sequencesearchingandalignmenthttps://speakerdeck.com/gregcaporaso/bio-299-lecture-8-10-nau-fall-2013#">Lecture 8-10 slides</a> I provide details on the differences between SW and NW initialization, scoring, and traceback.</p>
</div>
</div>
<div class="section" id="part-3">
<h3>Part 3<a class="headerlink" href="#part-3" title="Permalink to this headline">¶</a></h3>
<p>Define a general function that can perform global (Needleman-Wunsch; NW) or local (Smith-Waterman; SW) alignments.</p>
<p>Define a new function, <tt class="docutils literal"><span class="pre">generate_sw_and_traceback_matrices</span></tt> with the following form:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">generate_sw_and_traceback_matrices</span><span class="p">(</span><span class="n">seq1</span><span class="p">,</span><span class="n">seq2</span><span class="p">,</span><span class="n">gap_penalty</span><span class="p">,</span><span class="n">substitution_matrix</span><span class="p">)</span>
</pre></div>
</div>
<p>The return value should be the dynamic programming matrix and the traceback matrix for a SW alignment.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This will be much easier if you start with the <tt class="docutils literal"><span class="pre">generate_nw_and_traceback_matrices</span></tt> and modify it for Smith-Waterman.</p>
</div>
<p>Define a new function <tt class="docutils literal"><span class="pre">sw_traceback</span></tt> with the form:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">sw_traceback</span><span class="p">(</span><span class="n">traceback_matrix</span><span class="p">,</span><span class="n">sw_matrix</span><span class="p">,</span><span class="n">seq1</span><span class="p">,</span><span class="n">seq2</span><span class="p">)</span>
</pre></div>
</div>
<p>This function should return aligned the aligned sequences in the order they were passed in and the alignment score.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This will be much easier if you start with the <tt class="docutils literal"><span class="pre">nw_traceback</span></tt> and modify it for Smith-Waterman.</p>
</div>
<p>Next, define a new function <tt class="docutils literal"><span class="pre">sw_align</span></tt> with the form:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">sw_align</span><span class="p">(</span><span class="n">sequence1</span><span class="p">,</span><span class="n">sequence2</span><span class="p">,</span><span class="n">substitution_matrix</span><span class="p">)</span>
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This will be much easier if you start with your <tt class="docutils literal"><span class="pre">nw_align</span></tt> function and modify it for Smith-Waterman.</p>
</div>
<p>Define a new function <tt class="docutils literal"><span class="pre">align</span></tt> with the following form:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">align</span><span class="p">(</span><span class="n">sequence1</span><span class="p">,</span><span class="n">sequence2</span><span class="p">,</span><span class="n">substitution_matrix</span><span class="p">,</span><span class="n">local</span><span class="p">)</span>
</pre></div>
</div>
<p>Where <tt class="docutils literal"><span class="pre">local</span></tt> is a boolean (i.e., True or False) value. This function should return aligned_sequence1, aligned_sequence2, and the best alignment score. If <tt class="docutils literal"><span class="pre">local==False</span></tt>, an NW alignment should be performed. If <tt class="docutils literal"><span class="pre">local==True</span></tt> an SW alignment should be performed.</p>
<p>Run both local and global alignments as follows to test that this is working as expected:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">align</span><span class="p">(</span><span class="s">&#39;HEAGAWGHEE&#39;</span><span class="p">,</span><span class="s">&#39;PAWHEAE&#39;</span><span class="p">,</span><span class="n">blosum50</span><span class="p">,</span> <span class="bp">False</span><span class="p">)</span>
</pre></div>
</div>
<p>which should result in the following output:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">(</span><span class="s">&quot;HEAGAWGHE-E&quot;</span><span class="p">,</span> <span class="s">&quot;--P-AW-HEAE&quot;</span><span class="p">,</span> <span class="mf">1.0</span><span class="p">)</span>
</pre></div>
</div>
<p>and:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">align</span><span class="p">(</span><span class="s">&#39;HEAGAWGHEE&#39;</span><span class="p">,</span><span class="s">&#39;PAWHEAE&#39;</span><span class="p">,</span><span class="n">blosum50</span><span class="p">,</span> <span class="bp">True</span><span class="p">)</span>
</pre></div>
</div>
<p>which should result in the following output:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">(</span><span class="s">&quot;AWGHE&quot;</span><span class="p">,</span> <span class="s">&quot;AW-HE&quot;</span><span class="p">,</span> <span class="mf">28.0</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="guest-lecture-reports-due-11-february-2013-15-points-7-5-points-each">
<h2>Guest lecture reports (due 11 February 2013) (15 points; 7.5 points each)<a class="headerlink" href="#guest-lecture-reports-due-11-february-2013-15-points-7-5-points-each" title="Permalink to this headline">¶</a></h2>
<p>For each of the two guest lectures, turn in answers to the questions in <a class="reference external" href="https://docs.google.com/document/d/1Fieqfkbn-dMLjR6bpVqoT8E8Rb9HBtAfCSvesvIvLtI/edit">this document</a>. You can download this document and use it as a template for your assignment. You will turn these in as two separate PDFs by email to <a class="reference external" href="mailto:jc33&#37;&#52;&#48;nau&#46;edu">jc33<span>&#64;</span>nau<span>&#46;</span>edu</a>. Taking detailed notes during these lectures will make this assignment a lot simpler!</p>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<p class="last">Homework ids: <tt class="docutils literal"><span class="pre">johnson_lecture</span></tt> and <tt class="docutils literal"><span class="pre">butterfield_lecture</span></tt>; Extension: <tt class="docutils literal"><span class="pre">pdf</span></tt>; For this assignment, the files I turn in would be named <tt class="docutils literal"><span class="pre">jgc53_johnson_lecture.pdf</span></tt> and <tt class="docutils literal"><span class="pre">jgc53_butterfield_lecture.pdf</span></tt>.</p>
</div>
</div>
<div class="section" id="blast-exercises-due-4-february-2013-20-points">
<h2>BLAST exercises (due 4 February 2013) (20 points)<a class="headerlink" href="#blast-exercises-due-4-february-2013-20-points" title="Permalink to this headline">¶</a></h2>
<p>Using <a class="reference external" href="http://blast.ncbi.nlm.nih.gov/Blast.cgi?PROGRAM=blastn&amp;BLAST_PROGRAMS=megaBlast&amp;PAGE_TYPE=BlastSearch&amp;SHOW_DEFAULTS=on&amp;LINK_LOC=blasthome">NCBI nucleotide BLAST</a>, complete the <a class="reference external" href="https://docs.google.com/document/d/1x_ilvV9zW_SI1sFyqukhLz0Lnd4gAmwlVEJ4jrC814Q/edit">assignment worksheet</a>. You should turn in a PDF of that file with all answers filled in by email to <a class="reference external" href="mailto:jc33&#37;&#52;&#48;nau&#46;edu">jc33<span>&#64;</span>nau<span>&#46;</span>edu</a>.</p>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<p class="last">Homework id: <tt class="docutils literal"><span class="pre">blast</span></tt>; Extension: <tt class="docutils literal"><span class="pre">pdf</span></tt>; For this assignment, the file I turn in would be named <tt class="docutils literal"><span class="pre">jgc53_blast.pdf</span></tt>.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This assignment is derived from <a class="reference external" href="http://www.digitalworldbiology.com/BLAST/62000sequences.html">BLASTing Through the Kingdom of Life</a>. You may find <a class="reference external" href="http://www.digitalworldbiology.com/BLAST/index.html">this tutorial</a> to be very helpful.</p>
</div>
<p>Query sequences:</p>
<div class="highlight-python"><pre>&gt;Sequence1
AACAATTCATTTTTCCTGCTTTCCTAGAAAATTCTATAAAAGCTTCAAAA
TGAATTACTTGGTGATGATTAGTTTGGCACTTCTCTTCGTGACAGGTGTA
GAGAGTGTAAAAGACGGTTATATTGTCGACGATGTAAACTGCACATACTT
TTGTGGTAGAAATGCATACTGCAACGAGGAATGTACCAAGTTGAAAGGTG
AGAGTGGTTATTGCCAATGGGCAAGTCCATATGGAAACGCCTGTTATTGC
TATAAATTGCCCGATCATGTACGTACTAAAGGACCAGGAAGATGCCATGG
CCGATAAATTATAAGATGGAATGTATCCTAAGTATCAATGTTAAATAAAT
ATAATCAAAAAATT
&gt;Sequence2
CTAATAATCCTTGGAATACTCCTATATTTTGTATAAAGAAGAAATCAGGG
AAATGGAGAATGCTAATTGATTTTAGAGAACTTAATGCAAAAACAGAAAA
AGGAGCAGAAGTCCAATTAGGATTACCTCACCCATCTGGATTACAGAAGA
GAAAGAATGTAACAGTTTTAGATATAGGAGATGCTTATTTTACCATCCCT
TTAGATCCTGATTATCAGCCCTATACTGCATTTACTTTACCATCTAAGAA
TAATCAAAGTCCAGGAAAAAGGTATATTTGGAAATCTCTTCCACAGGGGT
GGGTCTTGAGTCCCTTAATATACCAGAGCACTCTAGATAATATTCTACAA
CCATTTAGAA
&gt;Sequence3
TCTTGGTGAGGATCCGTTGAGAACAACCCAACCGCCGCCCCATCGCCCTN
GTTAGANTNATGGCCGCGTCGGCGCTGCACCAGACCACCAGCTTCCTCNG
CACCGCCCCTCGCCGGGATGAGCTCGTCCGCCGCGTCGGCGACTCCGGTG
GCCGCATCACCATGCGCCGCACCGTCAAGAGCGCGCCCCAGAGCATCTGG
TATGGACCTGACCGTCCCAAGTNCCTGGGCCCGTTCTCGGAGCAGACGCC
ATCGTACCTGACCGGAGAGTTCCCGGGAGACTACGGGTGGGACACGGCGG
GGCTATCGGCCGACCCGGANACGTTCGCTATGAACAGGGAGCTGGANGTG
ATCCACTCNCGGTGGGCGATGCTGGGGGCGCTGGGCTGCGTCTTCCCGGA
GATCCTGTCCAANAACGGGG
&gt;Sequence4
GTTTTTAAAAGAGTTTGATCCTGGCTCAGGGTGAACGCGAATCAGCGCAC
TTAACACATGCAAGTTTTATGGATAGCTTTGAGCCTAGCTTTTAGTTAGA
CATAGCGAACGGGTGCGTAATGCTTAAGAATCTACTTTTAACTAAGGGAT
AATGGAAGGAAACTTTTGCTAATTCCTTATAGGTATGGATAAGATAACCT
ATCTTCATCTTGTTTAGAAAAAATTTGGCTGCTCAACGTAGTTAAGTTGG
TTAAAAAAGAGCTTGAATCTGATTAGTTAGTAGGTGAGGTAAAGGCTTAC
CTAGACGATAATCGGTAGCGGATCTGAGAGGATGACCCGCCACATTGGGA
CTGAGACACGGCCCAAACTTCTACGGAAGGCAGCAGTGAGGAATATTCTG
CAATGGGCGAAAGCCTGACAGTGTGACGCTGAGTGAAGGATGAAGGCCAC
AACCCGAGTTCGGGGGTCGTAAACTTCTTTTCCTAGGCGAAGAATAATGA
CTAACCTAGCAAGAAAGTATCGGCTAACTCCGTGCCAGCAGCCGCGGTAA
GACGGGGGATGCGAGCGTTATCCGGAATGACTGGGCGTAAAGCGTTTGTA
GGTGATCTTCTAAGTCTTGGTTTAAATCATAAAGCTTAACTTTTAAAAGA
GCCAAGATACTGGTTGAATAGAGTGAAATTGAGGTATTTGGGGGAATTCT
TAGAGGAATAGTAAAATGTAACGATACTAAGATGAAGACCGAAGGCGAAG
GCGTCATACTAAATTTTAACTGACACTCAAGGACGAAAGCTAGGGGAGCA
AATGGGATTAGAGACCCCAGTAGTCTTAGCAGTAAACGATGAGTACTAGA
TGTTGGACGCACGGTAATATATAATCTATTTATCTACTCGTTCGGTATCT
AAGCTAACGCAATAAGTACTCCGCCTGAGGAGTACGCTCGCAAGGGTGAA
ACTCAAAGGAATTGACGGGGGCTCGTACAAGCGGCGGAGCATGTGGTTTA
ATTTGATGCAAAGCAAAAAATCTTACCAGAGCTTGAAGTTGAAATTTTCA
AATTTAATCGATTTGAAAAGCCATAAATTGGCAAAAACGAGGTGGTGCAT
GGCTGTCGTCAGCTCGTGTCGTGAGACGTTGGGTTAATTCCCTTAACGAG
CGCAACCCTTGTCATAAGTTCTTTTGTCTTATGAGAAGGCTCGATTCGTC
GAGATTAAGAGGAGGATGACGTCAAGTCATCATGCCCTTTATGCTCTGGG
CTACACACGTGCTACAATGGTCGTTACAATAAGTACTGAAGAAAAAAACG
TATAACGATTATACGTTTAATTTTAACGTAAGTATGAAAATATTTACAAA
TCTTTAAAAGCGTAGCCCTAATATGAATCGTGGACTGAAACTCGTCCACG
TCAAACCGGAGTCGCTAGTAATCGCCGGTCACCATTACGGCGGTGAATAC
GTAACCGAGCCTTGTACACACCGCCCGTCACACCCTGGGAATTTAGGCTT
TTTGAAACATCTGCAGTGGGTGCGATTAAGGATTGGGTAACTGGGGTGAA
GTCGTAACAAGGTAGCGGTACTGGAAGGTGCAGCTGGA</pre>
</div>
</div>
<div class="section" id="gc-content-due-23-january-2013-10-points">
<h2>GC content (due 23 January 2013) (10 points)<a class="headerlink" href="#gc-content-due-23-january-2013-10-points" title="Permalink to this headline">¶</a></h2>
<p>Download a genome and compute its GC content. Copy or download <a class="reference external" href="https://docs.google.com/document/d/1iY1sfH9uKulmO0CLugtQOzBoAIGqh0oIwzZfa1ARay0/edit">the assignment</a>, fill in your answers, and turn the assignment in by email as a PDF.</p>
<p>Note that there are various ways that you can just look up the GC content, including via the IMG website. I&#8217;m asking you to compute it, and you&#8217;re being graded on your descriptions. Getting the right answer is a bonus (i.e., if you spend a couple of hours trying, and get it wrong, you&#8217;ll be graded on your well-documented effort, not your final answer).</p>
<p>Hints: Start with the IMG Genome Browser, and work with a bacterial, archaeal or viral genome.</p>
<p>Be creative - there are many ways to achieve this.</p>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<p class="last">Homework id: <tt class="docutils literal"><span class="pre">gc_content</span></tt>; Extension: <tt class="docutils literal"><span class="pre">pdf</span></tt>; For this first assignment, the file I turn in would be named <tt class="docutils literal"><span class="pre">jgc53_gc_content.pdf</span></tt>.</p>
</div>
</div>
</div>


      </div>
      <div class="bottomnav">
      
        <p>
        <a class="uplink" href="../../index.html">Contents</a>
        </p>

      </div>

    <div class="footer">
        &copy; Copyright 2012, J Gregory Caporaso.
      Last updated on Feb 18, 2013.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>